dali/operators/decoder/audio/audio_decoder_op.cc

```diff
@@ -59,14 +59,13 @@ AudioDecoderCpu::SetupImpl(std::vector<OutputDesc> &output_desc, const workspace
   }
   DALI_ENFORCE(IsType<uint8_t>(input.type()), "Raw files must be stored as uint8 data.");
   decoders_.resize(batch_size);
-  intermediate_buffers_.resize(batch_size);
   sample_meta_.resize(batch_size);
   files_names_.resize(batch_size);
 
-  decode_type_ = use_resampling_ || downmix_ ? DALI_FLOAT : output_type_;
-  TYPE_SWITCH(decode_type_, type2id, OutputType, (int16_t, int32_t, float), (
-      for (int i=0; i < batch_size; i++)
-        decoders_[i] = std::make_unique<GenericAudioDecoder<OutputType>>();
+  decode_type_ = (use_resampling_ && !downmix_) ? DALI_FLOAT : output_type_;
```

Comment:
So when we down mixing we don't care about the decode type itself?

Reply:
when we are resampling and not downmixing. Resampling expects `float`, downmixing can handle any type